drivers/amlogic/media/vout/hdmitx/hdmi_tx_20/hdmi_tx_main.c

```diff
@@ -5588,38 +5588,7 @@ static int hdmitx_set_current_vmode(enum vmode_e mode)
 
 static enum vmode_e hdmitx_validate_vmode(char *mode, unsigned int frac)
 {
-	struct vinfo_s *info = NULL;
-	struct hdmitx_dev *hdev = &hdmitx_device;
-
-	// force 4k50/60Hz to 420 unless manually set
-	if (strstr(hdev->fmt_attr, "rgb") == NULL &&
-	    strstr(hdev->fmt_attr, "420") == NULL &&
-	    strstr(hdev->fmt_attr, "422") == NULL &&
-	    strstr(hdev->fmt_attr, "444") == NULL) {
-		switch (hdev->hdmi_current_eotf_type) {
-			case EOTF_T_DOLBYVISION:
-			case EOTF_T_LL_MODE:
-			case EOTF_T_DV_AHEAD:
-				break;
-			default:
-				switch (hdmitx_edid_vic_tab_map_vic(mode)) {
-					case HDMI_3840x2160p50_16x9:
-					case HDMI_3840x2160p60_16x9:
-					case HDMI_4096x2160p50_256x135:
-					case HDMI_4096x2160p60_256x135:
-					case HDMI_3840x2160p50_64x27:
-					case HDMI_3840x2160p60_64x27:
-						if (!strstr(mode, "420"))
-							strncat(mode, "420", 3);
```

Comment:
I don't have a clear recollection of why this was added, perhaps this change is not needed anymore, but I'm pretty sure that it was done for compatibility reasons.
Here's my general observation from our experience with changes like these over the years.
"It works great here" is not an indication. It may work great with your setup, but will break for other people.

Reply:
With this code in place.  60hz HDR10 content appears to get stuck on 420, but bitrate set correctly.  I think this may be related to the previous fix.  The issue however from what I can tell by setting it here is this appears to trigger additional mode changes and in places that arguably shouldn't be happening--such as hitting fullscreen.

Reply:
I *think* that our thinking was that 420 is valid for video, and increases compatibility to bandwidth limited displays.